xds/src/main/java/com/linecorp/centraldogma/xds/internal/CentralDogmaSnapshotResources.java

```diff
@@ -84,14 +90,29 @@ private CentralDogmaSnapshotResources(
                 return collected.get(0).version();
             }
             if (collected.size() == versionedResources.size()) {
-                // All resources are included.
-                return allResourceVersion;
+                return allResourceVersion();
             }
 
             return Hashing.sha256().hashInt(collected.hashCode()).toString();
         };
     }
 
+    private String allResourceOrThrow(ResourceType resourceType) {
+        if (resourceType == ResourceType.CLUSTER || resourceType == ResourceType.LISTENER) {
+            return allResourceVersion();
+        }
+        throw new IllegalArgumentException("Requesting all resource isn't allowed for " + resourceType);
+    }
+
+    private String allResourceVersion() {
+        if (allResourceVersion != null) {
+            return allResourceVersion;
+        }
+        return allResourceVersion = Hashing.sha256()
+                                           .hashInt(versionedResources.values().hashCode())
```

Comment:
Note: Checked that the returned `values` for `ImmutableMap` will 1) consistently return the same hashcode if called on the same `versionedResources` 2) will return a hash of all elements which guarantees some degree of randomness.

Reply:
> 2) will return a hash of all elements which guarantees some degree of randomness.

Yeah, I also believe the chances of collision are extremely low given the xDS resources cardinality.
https://stackoverflow.com/a/4014407

Reply:
I don't doubt the randomness of SHA 256 😄    It's just that I've never thought about the hashcode of `ImmutableMap#values` before

Reply:
So did I before this PR. 😆